In [2]:
import pandas as pd
import numpy as np
import os
import sys
import matplotlib.pyplot as plt
import seaborn as sns

In [1]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("sewonghwang/golf-ball-testing")

print("Path to dataset files:", path)

100%|██████████| 598/598 [00:00<00:00, 598kB/s]

Extracting files...
Path to dataset files: C:\Users\KimDongyoung\.cache\kagglehub\datasets\sewonghwang\golf-ball-testing\versions\1


In [5]:
df = pd.read_csv(os.path.join(path, "Golf_test.csv"))
print("Dataframe shape:", df.shape)

Dataframe shape: (50, 6)


In [6]:
df.head()

TypeA_before  TypeA_after  TypeB_before  TypeB_after  TypeC_before  \
0           277          264           265          268           267   
1           269          261           264          267           264   
2           263          267           269          272           283   
3           266          272           274          277           287   
4           262          258           261          264           275   

   TypeC_after  
0          268  
1          269  
2          283  
3          266  
4          266

### Z-test 진행

Z-test를 선정한 이유
1. 연속형 확률분포
2. 표본 집단의 크기가 30이상일 때 사용
3. 단일 집단 (Type A)의 평균값 차이에 대한 분석이 목적


In [24]:
# 필요한 라이브러리 임포트
from statsmodels.stats.weightstats import ztest
from scipy import stats

In [10]:
# TypeA_before와 TypeA_after 컬럼 데이터 추출
typeA_before = df['TypeA_before'].dropna()  # 결측치 제거
typeA_after = df['TypeA_after'].dropna()    # 결측치 제거

In [18]:
mean_A = df['TypeA_after'].mean()
mean_B = df['TypeA_before'].mean()

print("TypeA_before 평균:", mean_B)
print("TypeA_after 평균:", mean_A)

TypeA_before 평균: 268.46
TypeA_after 평균: 270.72


In [19]:
# 두 집단 간 차이 계산
difference = typeA_before - typeA_after

# 차이의 표준 편차 계산
std_difference = difference.std()

# 결과 출력
print(f"두 집단 간 차이의 표준 편차: {std_difference}")

두 집단 간 차이의 표준 편차: 13.083421508433753


In [ ]:
# Z 값 계산

z_value = (mean_A - mean_B) / (std_difference / np.sqrt(len(typeA_before)))
print(f"Z 값: {z_value}")

# p-value 계산, p-value는 관찰된 데이터가 귀무가설 하에서 나타날 확률을 의미한다.
# 귀무가설: 두 집단 간 차이가 없다.
p_value = 1 - stats.norm.cdf(z_value)
print(f"p-value: {p_value}")

Z 값: 1.221439914972929
p-value: 0.1109597510944278


In [27]:
# Z-검정 수행
# 귀무가설: 두 그룹 간 평균 차이가 없다.
# 대립가설: 두 그룹 간 평균 차이가 있다.

# 유의 수준 설정 (예: 0.05)
alpha = 0.05
if p_value < alpha: # 유의 수준(alpha)보다 p-value가 작으면 귀무가설을 기각
    print("귀무가설을 기각합니다. 두 그룹 간 평균에 유의미한 차이가 있습니다.")
else:
    print("귀무가설을 기각하지 못합니다. 두 그룹 간 평균에 유의미한 차이가 없습니다.")

귀무가설을 기각하지 못합니다. 두 그룹 간 평균에 유의미한 차이가 없습니다.


p-value 값이 0.111로 유의 수준(0.05)보다 크기 때문에 귀무가설을 기각하지 못한다. 따라서 Type A 전과 후 평균에 유의미한 차이가 없다.